src/sentry/api/endpoints/artifact_lookup.py

```diff
@@ -0,0 +1,154 @@
+from sentry.api.endpoints.debug_files import has_download_permission
+from sentry.api.serializers import serialize
+from sentry.models import DebugIdArtifactBundle, File
+
+logger = logging.getLogger("sentry.api")
+
+
+@region_silo_endpoint
+class ArtifactLookupEndpoint(ProjectEndpoint):
+    permission_classes = (ProjectReleasePermission,)
+
+    def download_file(self, file_id, project):
+        rate_limited = ratelimits.is_limited(
+            project=project,
+            key=f"rl:DSymFilesEndpoint:download:{file_id}:{project.id}",
+            limit=10,
+        )
+        if rate_limited:
+            logger.info(
+                "notification.rate_limited",
+                extra={"project_id": project.id, "file_id": file_id},
+            )
+            return HttpResponse({"Too many download requests"}, status=403)
+
+        file = File.objects.filter(id=file_id).first()
+
+        if file is None:
+            raise Http404
+
+        try:
+            fp = file.getfile()
+            response = StreamingHttpResponse(
+                iter(lambda: fp.read(4096), b""), content_type="application/octet-stream"
+            )
+            response["Content-Length"] = file.size
+            return response
+        except OSError:
+            raise Http404
+
+    def get(self, request: Request, project) -> Response:
+        """
+        List a Project's Individual Artifacts or Bundles
+        ````````````````````````````````````````
+
+        Retrieve a list of individual artifacts or artifact bundles for a given project.
+
+        :pparam string organization_slug: the slug of the organization the
+                                          file belongs to.
+        :pparam string project_slug: the slug of the project to list the
+                                     DIFs of.
+        :qparam string debug_ids: If set, will query and return all the artifact
+                                  bundles that contain the given `debug_ids`.
+        :qparam string urls: If set, will query and return all the individual
+                             artifacts, or artifact bundles that contain files
+                             that match the `url`. This is using a substring-match.
+        :qparam string release: Used in conjunction with `urls`.
+        :qparam string dist: Used in conjunction with `urls`.
+
+        :auth: required
+        """
+
+        download_requested = request.GET.get("download") is not None
+        if download_requested and (has_download_permission(request, project)):
+            return self.download_file(request.GET.get("download"), project)
+        elif download_requested:
+            return Response(status=403)
+
+        # TODO: is there a better way to construct a url to this same route?
+        base_url = request.build_absolute_uri(request.path)
+
+        debug_ids = []
+        for debug_id in request.GET.getlist("debug_ids"):
```

Comment:
I'd call it `debug_id` instead, as we call it `query` in other filters, not `queries` and `checksum` not `checksums`, despite both being lists.